src/transports/tcp/TcpConnectionAcceptor.cpp

```diff
@@ -103,8 +103,12 @@ void TcpConnectionAcceptor::start(OnDuplexConnectionAccept onAccept) {
         for (auto& i : serverSocket_->getAddresses()) {
           LOG(INFO) << "Listening on " << i.describe();
         }
-      })
-      .get();
+      });
+
+  // XXX: Ideally we'd just do `future.get()`.  I can't explain why, but that
+  // consistently hits an asan-stack-overflow error in the exceptional case when
+  // built with gcc on Linux.  Doing `wait().getTry()` magically works.
+  future.wait().getTry().throwIfFailed();
```

Comment:
The old version of "fbcode/rsocket/transports/http2/HttpConnectionAcceptor.cpp" was having such a comment for the Future that's returned from server.start(). Though as we now don't return any Future, this code is deleted.

    // Need to synchronize on finishing HTTPServer::start() before ever calling
    // HTTPServer::stop().  Wait on either success or error callback to fire.  We
    // need to make a new future after waiting on the first one, otherwise the
    // calling .then() or .onError() fails with `setCallback called twice`.

    auto asTry = std::move(promise.getFuture().wait().getTry());
    return folly::makeFuture<folly::Unit>(std::move(asTry));

I am not sure if they are related but, as you see, they first wait and then do getTry and generate another future from the initial one.